pkg/server/pod_controller.go

```diff
@@ -247,36 +249,119 @@ func (c *PodController) markFailedPod(pod *api.Pod, startFailure bool, msg strin
 	}()
 }
 
-func (c *PodController) loadRegistryCredentials(pod *api.Pod) (map[string]api.RegistryCredentials, error) {
-	allCreds := make(map[string]api.RegistryCredentials)
-	for _, secretName := range pod.Spec.ImagePullSecrets {
-		s, err := c.resourceManager.GetSecret(secretName, pod.Namespace)
+func parseDockerConfigServer(serverURL string) (string, error) {
+	if !strings.HasPrefix(serverURL, "https://") &&
+		!strings.HasPrefix(serverURL, "http://") {
+		serverURL = "https://" + serverURL
+	}
+	parsed, err := url.Parse(serverURL)
+	if err != nil {
+		return "", err
+	}
+	return parsed.Host, nil
+}
+
+const (
+	dockerServerKey       = "server"
+	dockerUsernameKey     = "username"
+	dockerPasswordKey     = "password"
+	dockerAuthsKey        = "auths"
+	dockerConcatedAuthKey = "auth"
+	dockerConfigJSONKey   = ".dockerconfigjson"
+)
+
+func parseDockerConfigCreds(dockerJSON []byte) (map[string]api.RegistryCredentials, error) {
+	// dockerJSON looks like
+	// {
+	//     "auths": {
+	//         "https://index.docker.io/v1/": {
+	//             "username": "myuser",
+	//             "password": "mypass",
+	//             "email": "doesnt@matter.com",
+	//             "auth": "base64(username:password)"
+	//         }
+	//         "689494258501.dkr.ecr.us-east-1.amazonaws.com": {
+	//              "auth": "base64(username:password)"
+	//         }
+	//     }
+	// }
+	type dockerConfig struct {
+		Auths map[string]map[string]string `json:"auths"`
+	}
```

Comment:
It's a simple data structure, but in case you want use stricter typing: https://github.com/docker/cli/blob/master/cli/config/types/authconfig.go

Reply:
That works.